src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/ProgramDescriptor.java

```diff
@@ -32,13 +32,19 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.lang.module.InvalidModuleDescriptorException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
 import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.lang.model.SourceVersion;
+import jdk.internal.module.Checks;
```

Comment:
I'm trying to page in the build steps for jdk.compiler.interim. It this compiled with the boot JDK or the newly built JDK or both? I'm just wondering if this dependency on the internal class will be problematic or not, also wondering if it would be better for toPackageName test with SourceVersion.isIdentifier(name) instead.

Reply:
IIRC, `jdk.compiler` follows the N-1 rule, meaning that it and also ` jdk.compiler.interim` should be built with the boot JDK.

Reply:
>[...] also wondering if it would be better for toPackageName test with SourceVersion.isIdentifier(name) instead.

The initial commit of the PR does use `SourceVersion.isIdentifier(name)` and does not introduce an extra dependence on `jdk.internal.module`. See  https://github.com/openjdk/jdk/pull/19245/commits/ba9c886975ea8990774e237f6fdc03b441297e66 for details.

Reply:
We can't guarantee that jdk.internal.module.Checks will be stable across releases so dropping that dependency would be good.

Reply:
Even with `exports jdk.internal.module to jdk.compiler` in place?

https://github.com/openjdk/jdk/blob/52eda79522a5bd71b527e5946b654a331b021473/src/java.base/share/classes/module-info.java#L227-L231

Reply:
> Even with `exports jdk.internal.module to jdk.compiler` already in place before this change?

We can't have code in jdk.compiler that is compiled with both N-1 and N making use of classes in that internal package. So I think that will need to be fixed, for another PR of course, but let's not make it worse here.

Reply:
Reverted the second commit; back to use `javax.lang.model.SourceVersion` in order to test for valid package name elemnts.